src/components/Upload/Upload.tsx

```diff
@@ -581,18 +584,20 @@ const InternalUpload: React.ForwardRefRenderFunction<unknown, UploadProps> = (
             </Stack>
           )}
           {size === UploadSize.Medium && (
-            <Stack direction={'vertical'} gap={'m'}>
-              <Stack direction={'horizontal'} gap={'xs'}>
+            <Stack direction={'vertical'} flexGap={'m'}>
+              <Stack direction={'horizontal'} flexGap={'xs'}>
                 {renderIcon()}
                 {renderText()}
               </Stack>
               {renderButton()}
             </Stack>
           )}
           {size === UploadSize.Small && (
-            <Stack direction={'horizontal'} fullWidth justify={'space-evenly'}>
-              {renderIcon()}
-              {renderText()}
+            <Stack direction={'horizontal'} fullWidth justify={'space-between'}>
```

Comment:
I think you don't need the bindings to assign simple strings? true for above implementations as well
```suggestion
            <Stack direction="horizontal" fullWidth justify="space-between">
```

Reply:
Ya, bad habit of mine from angular days.

Reply:
Done